### DISCUSSION TITLE
__empty=False filter not working / filter on cabled front ports in Export Templates

### DISCUSSION THREAD

**Discussion by ajsiersema** [netbox-community/netbox#15350] [Q&A] [answered]
I'm trying to filter cabled / non cabled Device Front Ports in an Export Template.

This works for front ports **without** a cable:
```jinja2
{% for server in queryset %}
{%- for port in server.frontports.filter(cable_id=None)  -%}
```

Filtering ports **with** a cable, this seems to work, but leaves me with a nagging feeling that I really should be testing for not None (but how ?):
```jinja2
{%- for port in server.frontports.filter(cable_id__gt=-1)  -%}
```

I tried this instead, but I'm getting this error: `Unsupported lookup 'empty' for ForeignKey or join on the field not permitted.` :
```jinja2
{%- for port in server.frontports.filter(cable_id__empty=False)  -%}
```

Does anyone know if using `gt=-1` is the only solution, or if there's a better alternative ?

As an aside, `cabled` only seems to work in GraphQL:
```
device_list(name:"foo") {
  frontports(cabled: true)
```

**Comment by candlerb** [comment, marked as answer]
https://docs.djangoproject.com/en/4.2/ref/models/querysets/#field-lookups
https://docs.djangoproject.com/en/4.2/ref/models/querysets/#isnull

Looks like `cable_id__isnull=False` is likely to be the answer.

**Reply by ajsiersema** [reply]
Indeed.
Thanks for adding the links, very useful for any future query questions that might pop up, that are not covered by the Netbox documentation.

**Reply by candlerb** [reply]
Cheers. It's not always obvious, but the query parameters in REST API and GraphQL APIs (defined as "filtersets" in the code) are not the same as the filter expressions in the Django ORM, even though some terms are similar.
